collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -313,13 +311,67 @@ private static void logChoiceFilterAnalytics(QuestionDef question) {
                                     .send(new HitBuilders.EventBuilder()
                                     .setCategory("Itemset")
                                     .setAction(actionName)
-                                    .setLabel(formIdentifierHash)
+                                    .setLabel(getFormIdentifierHash())
                                     .build());
+
+                            if (predicate.toString().contains("current")) {
+                                return true;
+                            }
                         }
                     }
                 }
             }
         }
+        return false;
+    }
+
+    /**
+     * Gets a unique, privacy-preserving identifier for the current form.
+     *
+     * @return hash of the form title, a space, the form ID
+     */
+    private static String getFormIdentifierHash() {
+        String formIdentifier = "";
+        FormController formController = Collect.getInstance().getFormController();
+        if (formController != null) {
+            String formID = formController.getFormDef().getMainInstance()
+                    .getRoot().getAttributeValue("", "id");
+            formIdentifier = formController.getFormTitle() + " " + formID;
+        }
+
+        return FileUtils.getMd5Hash(
+                new ByteArrayInputStream(formIdentifier.getBytes()));
     }
 
+    /**
+     * Show an alert explaining the upcoming change in current() predicates.
+     */
+    private static void showCurrentPredicateAlert(Context context) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.current_predicate_warning_title);
+        builder.setMessage(R.string.current_predicate_warning);
+
+        DialogInterface.OnClickListener forumClickListener = (dialog, id) -> {
+            Intent intent = new Intent(context, WebViewActivity.class);
+            intent.putExtra("url", "https://forum.opendatakit.org/t/14557");
```

Comment:
This URL will be updated when we have a post up aimed at form designers with ways to address the change.

Reply:
Use 15122